### DISCUSSION TITLE
Dft_tools INP file

### DISCUSSION THREAD

**Discussion by AQuantumGuy** [TRIQS/dft_tools#214] [Q&A]
Dear TRIQS community,
I am stuck for my DMFT calculation. I am working with Titanium Dioxide (rutile phase) TiO2 and I've successfully run my calculations for Quantum Espresso and Wannier90. 

For Wannier90 I've calculated and plotted 22 Wannier bands which comply very similarly to QE results. I'm focusing on Ti-d and O-p bands both above and below the fermi level. Now I'm trying to write to INP file to run my calculations on TRIQS, but I'm not sure how to write it. I've seen from the documentation page, but I'm still confused, I'd appreciate if someone can help me write it.
The grid I've run my calculations is 14 14 14

Thank you for any help!

**Comment by phibeck** [comment]
Hi,

could you please paste what you have so far for the seedname.inp file, together with the projections card of your Wannier90 input? Just as a quick summary: In seedname.inp you can choose among those projections defined in your Wannier90 input. However, if you decide not to include any of those states in your DMFT calculations, they must be defined last in the Wannier90 projections.

**Reply by AQuantumGuy** [reply]
Hi, 
I actually don't know exactly how to construct the inp file. I know we describe the "specification of the k-mesh", "electron density", "number of atoms",  "atom, sort, l, dim, SO, irep", but I'm having troubles to understand what shall I input in these.

So far I understand that I shall input " 0    14 14 14" at the beginning.

My .win file code is:

`begin unit_cell_cart
ang
 	-0.0000000000  	-0.0000000000   	2.9692030000
  	4.6532720000  	-0.0000000000   	0.0000000000
  	0.0000000000   	4.6532720000   	0.0000000000
end unit_cell_cart

begin atoms_cart
ang
Ti       	0.5000000000   	0.5000000000   	0.5000000000
Ti       	0.0000000000   	0.0000000000   	0.0000000000
O        	0.5000000000   	0.1954200000   	0.8045800000
O        	0.5000000000   	0.8045800000   	0.1954200000
O        	0.0000000000   	0.3045800000   	0.3045800000
O        	0.0000000000   	0.6954200000   	0.6954200000
end atoms_cart

! system
num_wann = 22
num_bands = 70
mp_grid 14 14 14

! job control
iprint = 2

! plotting
wannier_plot = false
wannier_plot_supercell = 3
bands_plot = true
bands_num_points = 100
bands_plot_format = gnuplot
write_hr = true
write_u_matrices = true

! disentanglement
dis_win_min = 0
dis_win_max = 20
dis_num_iter = 1000
dis_conv_tol = 1.0E-12

! wannierisation
num_iter = 1000
conv_window = 5
conv_tol = 1e-12

! pp tools
kpath = false
kpath_task = bands
kpath_bands_colour = spin
kpath_num_points=500
begin kpoint_path
G  	0.00000  	0.00000  	0.00000 	X  	0.50000  	0.00000  	0.00000
X  	0.50000  	0.00000  	0.00000 	M  	0.50000 	-0.50000  	0.00000
M  	0.50000 	-0.50000  	0.00000 	G  	0.00000  	0.00000  	0.00000
G  	0.00000  	0.00000  	0.00000 	Z  	0.00000  	0.00000  	0.50000
Z  	0.00000  	0.00000  	0.50000 	R  	0.50000  	0.00000  	0.50000
R  	0.50000  	0.00000  	0.50000 	A  	0.50000 	-0.50000  	0.50000
A  	0.50000 	-0.50000  	0.50000 	Z  	0.00000  	0.00000  	0.50000
end kpoint_path

kslice = false
kslice_task = fermi_lines
fermi_energy = 8.6430
kslice_2dkmesh = 200 200
kslice_corner =  0.00  0.00  0.00
kslice_b1 = 	-0.25  0.25  0.25
kslice_b2 =  	0.25 -0.25  0.25

dos = true
dos_project = 1
!dos_energy_min = 8.
!dos_energy_max = 13.
dos_kmesh = 50
!dos_adpt_smr = false
!dos_smr_type = gauss
!dos_smr_fixed_en_width = 0.05

begin projections
Ti:d
O:p
end projections

begin kpoints`

**Comment by phibeck** [comment]
Please read this carefully https://triqs.github.io/dft_tools/latest/guide/conv_W90.html I do not know what you want to compute in the end. Based on your Wannier90 input, possible options are the following:

>  0  14 14 14               # specification of the k-mesh
some float                     # electron density (this is the filling you want to have. Probably partially filled Ti, perhaps you want to add oxygen?)
 2                      # number of atoms (or 6, if you want to add oxygen)
 0  0  2  5  0  0       # atom, sort, l, dim, SO, irep (d-shell corresponds to l=2, all 5 orbitals, so dim=5)
 1  0  2  5  0  0       # atom, sort, l, dim, SO, irep (I assume here that this Ti atom is symmetry-equivalent to Ti atom 1 (sort=0), verify!)
*** the next four lines you might add if you want to include the oxygen p states
 2  1  1  3  0  0       # atom, sort, l, dim, SO, irep (p-shell corresponds to l=1, all 3 orbitals, so dim=3)
 3  1  1  3  0  0       # atom, sort, l, dim, SO, irep (I assume here that all O atoms are symmetry-equivalent to O atom 1 (sort=1), verify!)
 4  1  1  3  0  0       # atom, sort, l, dim, SO, irep
 5  1  1  3  0  0       # atom, sort, l, dim, SO, irep

Please specify if anything is unclear.

**Reply by AQuantumGuy** [reply]
Many thanks for this and apologies for my continuous requests. 
The output I get when I run the Wannier90Converter() code, I get the following error:

```
Reading input from tio.inp...
Total number of WFs expected in the correlated shells: 22
Number of inequivalent shells: 2
Shell representatives: [0, 2]
Mapping: [0, 0, 2, 2, 2, 2]

The Hamiltonian in MLWF basis is extracted from tio files...
reading tio_hr.dat       

... done: 3375 R vectors, 22 WFs found
Number of WFs equal to number of correlated orbitals
H(R) has large complex components at R 0
H(R) has large complex components at R 1
....
....
H(R) has large complex components at R 3374
ERROR: eigenvalue mismatch between equivalent shells! 1
Eigenvalue difference [3.4114 2.2971 0.6848 1.1158 0.7937], but threshold set to 2.0e-06.

Consider lowering threshold if you are certain the mapping is correct.
ERROR: rot_mat does not map H(0) correctly! 1
ERROR: eigenvalue mismatch between equivalent shells! 3
Eigenvalue difference [-0.6166  1.6379  3.1782], but threshold set to 2.0e-06.
Consider lowering threshold if you are certain the mapping is correct.
ERROR: rot_mat does not map H(0) correctly! 3
ERROR: eigenvalue mismatch between equivalent shells! 4
Eigenvalue difference [ 0.2249 -0.1579 -1.7399], but threshold set to 2.0e-06.
Consider lowering threshold if you are certain the mapping is correct.
ERROR: rot_mat does not map H(0) correctly! 4
ERROR: eigenvalue mismatch between equivalent shells! 5
Eigenvalue difference [ 0.0145 -0.2404 -1.9619], but threshold set to 2.0e-06.
Consider lowering threshold if you are certain the mapping is correct.
ERROR: rot_mat does not map H(0) correctly! 5



AttributeError:   Traceback (most recent call last)

/tmp/ipykernel_4138/655405481.py in <module>
      1 get_ipython().run_line_magic('cd', '"./QE/mlwf"')
      2 Converter = Wannier90Converter(seedname='tio', hdf_filename='tio_wannier.h5')
----> 3 Converter.convert_dft_input()
      4 get_ipython().run_line_magic('cd', '"../../"')
      5 # and copy the h5 archive in the cwd

/usr/lib/python3.10/dist-packages/triqs_dft_tools/converters/wannier90.py in convert_dft_input(self)
    385             # of the first spin
    386             if isp == 0:
--> 387                 use_rotations, rot_mat = self.find_rot_mat(
    388                     n_corr_shells, corr_shells, shells_map, ham_corr0)
    389             else:

/usr/lib/python3.10/dist-packages/triqs_dft_tools/converters/wannier90.py in find_rot_mat(self, n_sh, sh_lst, sh_map, ham0)
    877 
    878         # abort in case the rot_mat was not found correctly to avoid the user to ignore ERRORS
--> 879         if not succeeded: mpi.MPI.COMM_WORLD.Abort(1)
    880 
    881         return succeeded, rot_mat

Attribute error: module ‘triqs.utility.mpi’ has no attribute ‘MPI’
```

**Reply by phibeck** [reply]
This just tells you that none of your shells are symmetry-equivalent. It failed to map the shells you marked as symmetry-equivalent. You would have to modify the input to:

> 0 14 14 14
some float
6
0 0 2 5 0 0
1 1 2 5 0 0
2 2 1 3 0 0
3 3 1 3 0 0
4 4 1 3 0 0
5 5 1 3 0 0

Note that this works always, but disregards any symmetries of the system. You have to think about whether this is correct and what the symmetries of your system are. You can start by comparing the energy levels of the individual sites (in `seedname_hr.dat` as well as their spread from `seedname.wout`. If the mismatch in the mapping is small, sometimes this can also occur if one uses too few kpoints. But this does not seem to be the case here.

**Reply by AQuantumGuy** [reply]
Hi, my calculations mostly work whilst following the steps of the [tutorial](https://github.com/wannier-developers/wannier-tutorials/blob/master/2022_05_Trieste/DAY3_AM_TRIQS/triqs_tutorial_solution.ipynb).

I just got one difference, if you could just let me know what can be the reason, I'd really appreciate that. The difference as you can see from the [Graph](https://drive.google.com/file/d/17EFkNNtVSkTzAwqoqa_oNCmEMUc8ZP6o/view?usp=sharing), is the red lines from my W90 code. The TRIQS calculation lines (in white) are similar to W90 but are spread to a small frame (as in that these look similar at different k-points however the band gap is small and are spread to small energy levels)

Sincerely,

**Reply by AQuantumGuy** [reply]
Hi, just for a reminder. Would you know the reason on why the TRIQS calculations are "scaled" smaller than W90 one? Is it an error in my code?

**Reply by the-hampel** [reply]
Hi, 
before we try to analyze this further it might be a good idea to retry the postprocessing step (no need to rerun solid_dmft, or do the analytic continuation again) with the latest version of the [plot_correlated_bands.py](https://github.com/TRIQS/solid_dmft/blob/unstable/python/solid_dmft/postprocessing/plot_correlated_bands.py) script of solid_dmft on the unstable branch: https://github.com/TRIQS/solid_dmft/tree/unstable . Could you please try this version? 

Since the unstable version of solid_dmft is not compatible with the 3.1.x version of dft_tools you have to be a bit careful. Maybe the easiest way is to just download this one file [plot_correlated_bands.py](https://github.com/TRIQS/solid_dmft/blob/unstable/python/solid_dmft/postprocessing/plot_correlated_bands.py) and replace the current version of it in the installation directory. If it worked you should notice immediately that some of the input parameters of the `get_dmft_bands` function have been renamed. Let me know if you have problems using this.

**Reply by AQuantumGuy** [reply]
Hi, just tried updating the script for plotting, but I still get the same result. Like I said, the band lines perfectly resemble W90 ones, but are scaled smaller and I can't find out what might be the reason. Any suggestion would be appreciated!

Sincerely,

**Reply by phibeck** [reply]
Okay. For the updated plotting script you should have encountered some issues that have to do with renaming variables. Can you confirm you encountered these so we can be sure the updating of the plotting script worked?

Could you please share an image of the self-energy you used (should be on the real frequency axis), and remove the tag `w_mesh` in the dictionary that contains the information about the self-energy, then try again?

**Reply by AQuantumGuy** [reply]
Hi, I just realized that it did not update the file and I've just manually updated it and as Alexander stated, I now do get error stating missing argument "mu_tb". Whats the best way to proceed on this?

Sincerely,

**Reply by the-hampel** [reply]
Yes now you have the newest version. You have to change in the `w90_dict` the name of the paramter `mu` to `mu_tb`. Also there are some other changes. The function now returns only 3 values. Please remove the `dft_mu` output value. You can find a working example here: https://triqs.github.io/solid_dmft/tutorials/correlated_bandstructure/plot_correlated_bands.html . If the result looks still weird, consider adding the paramter `add_mu_tb=True` to the `get_dmft_bands` function call.

**Reply by AQuantumGuy** [reply]
Hi, tried following the advice, I haven't got the plot, but now I receive the following error:
_Number of Wannier orbitals: 22 and self-energy target_shape (1, 1) does not match_

The self-energy graph is shown [here](https://drive.google.com/file/d/1Ok7JkX8Cgd4WP5IrLH0DNSGC4FmuTgag/view?usp=sharing) and 
[here](https://drive.google.com/file/d/1cGR-5kNbHpj9GsOolneC-Ujc3MKFcGt6/view?usp=sharing)
From my W90 calculations, I do obtain 22 bands. 

Sincerely,

**Comment by phibeck** [comment]
Hi, okay so this clarifies the issues better. Currently, there seem to be at least two problems:

1. your self-energy is computed for a single orbital, while the program expects a 22x22 block for this to work. Did you do your DMFT calculations for all 22 bands or only a subset of these? The self-energy needs to have the same shape as the number of Wannier functions. So in case you have multiple blocks or impurities, they will have to be merged. We can discuss how this can be achieved, but we first need to know what you did for the DMFT calculations. Perhaps you could upload the output of your DMFT calculation, that would be helpful.
2. the plot of your self-energy is a Matsubara self-energy, i.e. on the imaginary axis. Did you use analytic continuation? I suppose the answer is yes, otherwise the code should have triggered another error. Could you show a plot of the real-frequency self-energy?

**Reply by AQuantumGuy** [reply]
Hi,
By DMFT output do you mean the "dmft.out" file we get when running "mpirun solid_dmft"? If you mean that, then it is attached [here](https://drive.google.com/file/d/17VXAHW2cfVR5GMOhO0i1d1VLfdmBjNEb/view?usp=sharing). 

If I'm not wrong, the real-part of the self-energy should be [this](https://drive.google.com/file/d/1gqi9r4ghXdbm7OKhB0ReHVFhlSuewj9m/view?usp=sharing) one. (I've done this by putting _ax.oplot(Sigma_imp_iw['up_0'].real, '-o'_ )

Truely appreciate the help of both of you!

**Reply by the-hampel** [reply]
Hi, 
ah yes for this complicated set of orbitals and impurities there is a lot of manual fixing involved at the moment when using the postprocessing tool. We would need to fix that first. I am still confused about the self-energy. This is still a Matsubara self-energy (you see iw_n on the x-axis label). It is the real part of it, but we need it on the real frequency axis! Did you do the analytic continuation? If so you should find in the h5 archive under `last_iter` a new entry called something like `Sigma_MaxEnt` ? We would like to see this object. 

The thing in the postprocessing I can try to fix in the near future. If you anyway want to try it you have to assemble one large 22x22 self-energy from all the blocks you obtained, which is maybe a bit too complicated?

**Reply by AQuantumGuy** [reply]
Hi, I understand and can look into the self-energy from the archive. The analytical continuation of self-energy is shown [here](https://drive.google.com/file/d/1TXwRJO34jnVSMpckWS15MNJjBxrHl6Dh/view?usp=sharing)

I am actually willing to obtain the large 22x22 self-energy as I need it for my studies (I understand it must take long processing time as well). A direction/insight from you is so much appreciated with the steps to take. 

Sincerely,

**Reply by the-hampel** [reply]
Okay, maybe we start working on the analytical continuation. I can meanwhile try to fix the plotting script to work with multiple impurities etc. 

Your continue self-energy looks a bit weird to me. The imaginary part is completely 0 for omega>0 . How did you run the analytic continuation? Can you show the output please?

**Reply by phibeck** [reply]
This also just shows one orbital, but based on your DMFT output you are going to need 22 analytically continued self-energies. How do the others look?

**Reply by AQuantumGuy** [reply]
The analytical continuation output is [this](https://drive.google.com/file/d/11Q3ksreZDZychvNFVgQ8z_GE_fC71Upf/view?usp=sharing).

The others orbital  are not much different, all of them show the imaginary part as 0 for omega > 0.

When I run "list(Sigma_imp_iw.indices)", it shows 
'up_0'
'up_1'
'up_2'
'up_4'
'down_0'
'down_1'
'down_2'
'down_3'
'down_4'

**Reply by the-hampel** [reply]
I think we have to tune the analytic continuation a bit, and maybe even obtain the Sigma(iw) with higher precision in the last iteration. You can try the following: 
1) change in the sigma_maxent error by increasing `maxent_error` a bit when calling it. Did you use `0.03`? Try to increase to `0.05` and see if the optimal alpha value goes up a bit (currently 17)
2) change `continuator_type='inversion_sigmainf'` to `continuator_type='inversion_dc'`

If the result look still the same, you should maybe try to rerun you DMFT calculation for 1 additional iteration on top of the old one, as it is done in the tutorial https://github.com/TRIQS/tutorials/blob/unstable/AbinitioDMFT/solutions/01s-solid_dmft.ipynb . Increase the number of cycles by one magnitude and put n_iter_dmft=1. Then try again the Sigma Maxent. If they still look the same this can be also a good sign, meaning that your analytical continuation is quite robust. But it will be good to check this.

**Reply by AQuantumGuy** [reply]
I tried changing the sigma_maxent error to 0.05 and the optimal alpha value now goes to 21, but the result still looks a bit weird. See it [here](https://drive.google.com/file/d/1jC8JmwMmDpEy3RT5DURrmXlUwdpcgDNd/view?usp=sharing).

I tried to rerun the DMFT calculations, however, in the output file I just see the following error:
_TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''_
This is shown right at the beginning of the output file and then I don't see any calculation.

**Reply by AQuantumGuy** [reply]
UPDATE: I tried to modify the _seedname.inp_ by not putting the Fermi energy at the end of the file. This time, the self-energy plots look different and contain more lines. Could you check if this [plot](https://drive.google.com/file/d/1rCorwfRj0pZFKtPuaYMDNM9t0tFkUlOM/view?usp=sharing) looks more correct?

The output from DMFT calculation is [this](https://drive.google.com/file/d/1xyy99brYnxXkZ1dhDtN533NIpVfKaWEY/view?usp=sharing).


However, when plotting the spectral function, now it gives the following error:
_Number of Wannier orbitals: 22 and self-energy target_shape (5, 5) does not match_



Is there any reason that when I don't put fermi energy, I get much different plots? Besides that, how could I proceed to plot correct spectral function?

Sincerely,

**Reply by the-hampel** [reply]
Hi yes the self-energies look much better now. However, you should carefully check if you expect a complex result. Ultimately you have to decide if the result is correct or not. 

It is clear that you still have the error. I cannot guarantee when I have time to fix this. If you need a results earlier you need to create a larger self-energy with 22 orbitals and fill in the self-energy of your system manually. Your system is quite complicated with many impurities. You have to figure this out on your own. The post-processing scripts expects at the moment that you provide a self-energy with a shape matching the number of Wannier orbitals. I will update you when I found time to fix this on our side. 

Best,
Alex

**Reply by AQuantumGuy** [reply]
I'm not sure which part do I need to modify to "merge" and having this large self-energy with 22 orbitals

**Reply by phibeck** [reply]
In essence you will have to construct an artificial 22x22 sized Green's function, where on the diagonal entries you place the analytically continued 1x1 self-energies for each orbital of all of your states. Please see here for basic instructions how to construct and manipulate Block Green's functions: https://triqs.github.io/triqs/unstable/userguide/python/tutorials/Basics/solutions/01s-Greens_functions.html#Block-Green-functions

A few side notes:
- all impurities and orbitals must have the same order as in your DMFT calculations for this to make sense
- you will have to write this into the hdf5 archive, or pass the Sigma directly to the plotting routine

Regarding the Fermi energy at the end of the `seedname.inp` file: In principle there should be no difference between your self-energies. Did you check whether your DMFT solution is the same for both calculations?

**Reply by AQuantumGuy** [reply]
Hi, I tried reading the tutorial for manipulating Green's function. Unfortunately, I haven't really understood how shall I modify in my specific case with my data. 

I tried re-running my simulations again with W90 and focusing only on d-states and having only 10 bands. My self-energy plot now looks like [this](https://drive.google.com/file/d/1g08GmiHmU1eUWbhPg7eluQEa2uue_IW3/view?usp=sharing). 
I still have the error for spectral function Number of Wannier orbitals: 10 and self-energy target_shape (5, 5) does not match.

I understand it requires a shape of 10x10 now, but I do not understand much how to modify. 
Before I updated the plot_correlated_bands.py file, it did actually show the spectral function, but it was scaled smaller. Is there not any other way I can fix this?

I tried to plot using the previous code of plot_correlated_bands.py for only "d-states" and get the following [graph](https://drive.google.com/file/d/1p2dPWG7M5XubbSToxlUHJwlfMKfKP4ZV/view?usp=sharing)

Sincerely,

**Reply by AQuantumGuy** [reply]
With old plot_correlated_bands.py and NOT putting Fermi energy in seedname.inp file, I get this new error when plotting spectral function:
Operands could not be broadcast together with shapes (1000, 5, 5)   (10, 10)

**Reply by the-hampel** [reply]
Which part of constructing a larger BlockGf is not clear? Maybe this helps: Imagine you want to create a BlockGf `S_w_upfold` with two blocks, each having a target_shape of 5x5 (or whatever number of wannier orbitals you have in a wannier90_hr.dat) and inheriting the mesh of a continued self-energy `S_w_cont` with shape 2x2. Let's first create the large target BlockGf:
```
prototype = GfReFreq(mesh= S_w_cont.mesh, target_shape=[5,5])  # creates an 0.0 initialized Gf object with shape 5x5
S_w_upfold = BlockGf(name_list = ['up_0', 'down_0'], block_list = [prototype, prototype], make_copies = True)
```
Let's assume in our example that the first 2x2 block of up/down should be diagonal and degenerate holding information from `S_w_cont` from the first impurity of your DMFT calculation matching the first two orbitals in your wannier90 setup. We can fill the first 2x2 blocks of `S_w_upfold` with the following lines:
```
for block, gf in S_w_upfold:
    S_w_upfold[block][0,0] << S_w_cont[block][0,0]
    S_w_upfold[block][1,1] << S_w_cont[block][1,1]
```
Now the first 2 orbitals in `S_w_upfold` are filled with the continued self-energy and the rest is 0.0. If you have a second impurity that is equivalent to the first and has not been treated by solid_dmft, i.e. the next 2 wannier orbitals are the same but belong to a different site, you can also duplicate the self-energies to these obitals:

```
for block, gf in S_w_upfold:
    S_w_upfold[block][2,2] << S_w_cont[block][0,0]
    S_w_upfold[block][3,3] << S_w_cont[block][1,1]
```
The fifth orbital of `S_w_upfold` in this example is empty and can represent uncorrelated wannier states in your system.

This is exactly what you need to do, but just with different number of orbitals. Additionally you might need to subtract some double counting from some orbitals. The `get_dmft_bands` function will automatically subtract the DC from DMFT for ALL orbitals in Sigma. So you might need to add it to some of your orbitals.

I hope this helps a bit. Please let me know which part of creating a larger self-energy is unclear. I will also update you here once I fixed the script to load all information from the archive automatically. For now this has to be done manually. 
- Alex

**Reply by AQuantumGuy** [reply]
So from what you explained, first you created a BlockGF with two blocks both of size 5x5. Then you focused only on the first block and you filled diagonally until the 4th orbitals and leaving 5th as zero.

Since the error I get is "Number of Wannier orbitals: 10 and self-energy target_shape (5, 5) does not match.", I need to create a target_shape (10, 10), right?


When I run "Sigma_imp_iw" I get the following (**WHEN I DON'T PUT FERMI ENERGY in seedname.inp file**)
```
Green Function Sigma_freq_0 composed of 2 blocks: 
 Greens Function Sigma_freq_0_up_0 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (5, 5): 
 
 Greens Function Sigma_freq_0_down_0 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (5, 5): 

```


From what I understand, I do need to construct a 10x10 GF where in the diagonal entries I place the analytically continued 1x1 self-energies for each orbital of all my states. When I run "Sigma_imp_iw" I get the following (**WHEN I PUT FERMI ENERGY in seedname.inp file**)

```
Green Function Sigma_freq_0 composed of 10 blocks: 
 Greens Function Sigma_freq_0_up_0 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (1, 1): 
 
 Greens Function Sigma_freq_0_up_1 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (1, 1): 

 Greens Function Sigma_freq_0_up_2 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (1, 1): 
 
 Greens Function Sigma_freq_0_up_3 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (1, 1): 

 Greens Function Sigma_freq_0_up_4 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (1, 1) : 
 
 Greens Function Sigma_freq_0_down_0 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (1, 1): 

 Greens Function Sigma_freq_0_down_1 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (1, 1): 

 Greens Function Sigma_freq_0_down_2 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (1, 1): 

 Greens Function Sigma_freq_0_down_3 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (1, 1): 

 Greens Function Sigma_freq_0_down_4 with mesh Matsubara Freq Mesh of size 502, Domain: Matsubara domain with beta = 10, statistic = Fermion, positive_only : 0 and target_shape (1, 1): 
```


So you can see that I do actually get the GF that you guys suggested (when putting Fermi energy) and plotting using the previous plot_spectral function code. Just a reminder, when plotting the spectral functions, these are a bit different from W90 ones. Example, [here ](https://drive.google.com/file/d/1bsYzMJDlxbJsJ0dJ0Bf_DJAx2Nw8eigC/view?usp=sharing)is the result when I do only 3 dmft_iteractions.

**Reply by the-hampel** [reply]
This GF has 10 1x1 blocks instead of 1 10x10 block. I do not see what the Fermi flag in the w90 converter has to do with that? btw. you can use it either way and specify this later by passing to get_dmft_bands the following flag `add_mu_tb=False/True` . Anyway, it seems everything is now working? Or do you have a question?

**Reply by AQuantumGuy** [reply]
I understand, but considering still where I need to modify my GF to make it 10x10GF. I do need to construct a 10x10 GF, however since I have a 5x5 block of "up_0" and "down_0", how do I fill the rest of the data of 5x5 to fill the whole 10x10 GF?

When I run the analytical continuation, the first few lines are:
```
Finished reading h5 archive. Found 2 impurities.
Created sigma continuator of type "inversion_sigmainf"
Starting run of maxent now.
Continuing impurities with blocks:
- Imp 0: ['up_0', 'down_0']
- Imp 1: ['up_0', 'up_1', 'up_2', 'up_3', 'down_0', 'down_1', 'down_2', 'down_3' ]
- Appending
- Appending
- .....

```

Does this mean that I need to construct 10x10 GF with the first two orbitals for all 'up_' cases? If that is actually the case, could you remind me how can I access and input Imp 0 data (up_0) in first two block then moving into Imp 1 (up_0) for next position, etc.

**Comment by the-hampel** [comment]
Hi @AQuantumGuy ,

We adapted the postprocessing script of solid_dmft to take care of the transformation from impurity to lattice self-energy in case the user uses the wannier90 interface without entanglement:  
https://github.com/TRIQS/solid_dmft/blob/DEV_PCB/python/solid_dmft/postprocessing/plot_correlated_bands.py 
please copy this file and replace your local version of it. Then you can use the script as described in the tutorial without changes and it should automatically convert the Sigma stored in the h5 from the Maxent run to the correct blockstructure matching the size of the wannier90_hr.dat file.
